awscrt/http.py

```diff
@@ -112,6 +112,26 @@ def new(cls,
             If successful, the Future will contain a new :class:`HttpClientConnection`.
             Otherwise, it will contain an exception.
         """
+        return HttpClientConnection._generic_new(
+            host_name,
+            port,
+            bootstrap,
+            socket_options,
+            tls_connection_options,
+            proxy_options)
+
+    @staticmethod
+    def _generic_new(
```

Comment:
nit: ignore if this has been discussed already. generic name seems a bit weird. maybe base or common or something?